meta/system.py

```diff
@@ -116,12 +132,13 @@ def add_overlay(self, origin, name):
 
         self.jobs[name] = {}
 
-    def copy_from_system(self, origin_system, origin_name, target_name=None):
+    def copy_from_system(self, origin_system: "System", origin_name: str, target_name: Optional[str] = None):
```

Comment:
Can you try `from __future__ import annotations` to solve the System not being defined problem?